pkg/xds/generator/egress_generator.go

```diff
@@ -0,0 +1,60 @@
+package generator
+
+import (
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	model "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+)
+
+const (
+	EgressProxy = "egress-proxy"
+
+	// OriginEgress is a marker to indicate by which ProxyGenerator resources were generated.
+	OriginEgress = "egress"
+)
+
+type EgressGenerator struct {
+}
+
+func (g EgressGenerator) Generate(ctx xds_context.Context, proxy *model.Proxy) (*model.ResourceSet, error) {
+	resources := model.NewResourceSet()
+
+	listener, err := g.generateLDS(proxy, proxy.ZoneEgressProxy.ZoneEgressResource, proxy.APIVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	resources.Add(&model.Resource{
+		Name:     listener.GetName(),
+		Origin:   OriginEgress,
+		Resource: listener,
+	})
+
+	return resources, nil
+}
+
+// generateLDS generates one Ingress Listener
+// It assumes that mTLS is on. Using TLSInspector we sniff SNI value.
+// SNI value has service name and tag values specified with the following format: "backend{cluster=2,version=1}"
+func (g EgressGenerator) generateLDS(
+	proxy *model.Proxy,
```

Comment:
proxy is not used

Reply:
I know as this is just a temporarily dummy function, which will be implemented in the next task